### PR TITLE
Deployment module

### PR DESCRIPTION
**Pull request by simonduq**
This PR contributes a deployment module, that helps manage mappings between node-id and MAC/IPv6 addresses. This is mostly intended for deployments and testbed runs, where a static list of nodes and their MAC address is known in advance.

To use, the use simple provides a map of all node IDs and their MAC address. For instance, to deploy in Flocklab, Openmote platform, use:
```
const struct id_mac deployment_flocklab_openmotes[] = {
  {  3, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0x29}}},
  {  6, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0x34}}},
  {  8, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0x1f}}},
  { 15, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0x85}}},
  { 16, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0x00}}},
  { 18, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0x37}}},
  { 22, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0x08}}},
  { 23, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0x5f}}},
  { 31, {{0x00,0x12,0x4b,0x00,0x06,0x0d,0x9b,0xb1}}},
  {  0, {{0}}}
};
```

And:
```
#define DEPLOYMENT_MAPPING deployment_flocklab_openmotes
```

This module will automatically set the node ID at each node, consistently with the testbed IDs (i.e., what the testbed infrastructure refers to as node 3 will know it is node 3).

From there, a number of functions are provided, to check the total node count, or convert IDs to LLADDR to 6ADDR and back.

When the module is used, the `LOG_CONF_WITH_COMPACT_ADDR` now operates differently. It will transform all address logs so as to show which node ID the address belongs to. Below is an example output, for the RPL link table:
```
ID:1	[INFO: RPL       ] links: 8 routing links in total (Periodic)
ID:1	[INFO: RPL       ] links: 6G-001  (DODAG root) (lifetime: infinite)
ID:1	[INFO: RPL       ] links: 6G-003  to 6G-001 (lifetime: 1680 seconds)
ID:1	[INFO: RPL       ] links: 6G-002  to 6G-001 (lifetime: 1680 seconds)
ID:1	[INFO: RPL       ] links: 6G-005  to 6G-003 (lifetime: 1680 seconds)
ID:1	[INFO: RPL       ] links: 6G-004  to 6G-002 (lifetime: 1680 seconds)
ID:1	[INFO: RPL       ] links: 6G-007  to 6G-004 (lifetime: 1680 seconds)
ID:1	[INFO: RPL       ] links: 6G-008  to 6G-005 (lifetime: 1680 seconds)
ID:1	[INFO: RPL       ] links: 6G-006  to 6G-005 (lifetime: 1680 seconds)
ID:1	[INFO: RPL       ] links: end of list
```

Here, `6G` means "Global IPv6", but we also add `6L` ("link-local IPv6"), `6M` ("multicast IPv6"), and `LL` ("link-layer address").

The PR adds an example under `libs/deployment`, and a Travis compile test for it.

### CODE REVIEW

**Comment by g-oikonomou** [comment]
World-first 6G :P

**Comment by simonduq** [comment]
> World-first 6G :P

We're next-gen for a reason ;)

File: os/services/deployment/deployment.c [hunk 1]
```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2018, Swedish Institute of Computer Science.
```
Comment (joakimeriksson) [inline os/services/deployment/deployment.c]: Love that! Computer Science 4ever!

Comment (simonduq) [reply os/services/deployment/deployment.c]: ah good catch :p I'll fix and rebase :)

**Pull request merged**